core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -298,78 +312,69 @@ class RequestChannel(val numProcessors: Int, val queueSize: Int) extends KafkaMe
 
   def updateErrorMetrics(apiKey: ApiKeys, errors: collection.Map[Errors, Integer]) {
     errors.foreach { case (error, count) =>
-      requestChannelMetrics.markErrorMeter(apiKey.name, error, count)
+      metrics(apiKey.name).markErrorMeter(error, count)
     }
   }
 
   def shutdown() {
     requestQueue.clear()
-    requestChannelMetrics.shutdown()
+    metrics.shutdown()
   }
 
   def sendShutdownRequest(): Unit = requestQueue.put(ShutdownRequest)
-
-}
-
-class RequestChannelMetrics {
-
-  private val metricsMap = mutable.Map[String, RequestMetrics]()
-
-  (ApiKeys.values.toSeq.map(_.name) ++
-      Seq(RequestMetrics.consumerFetchMetricName, RequestMetrics.followFetchMetricName)).foreach { name =>
-    metricsMap.put(name, new RequestMetrics(name))
-  }
-
-  def requestMetrics(metricName: String) = metricsMap(metricName)
-
-  def markErrorMeter(name: String, error: Errors, count: Int) {
-      val errorMeter = metricsMap(name).errorMeters(error)
-      errorMeter.getOrCreateMeter().mark(count.toLong)
-  }
-
-  def shutdown(): Unit = {
-     metricsMap.values.foreach(_.removeMetrics())
-  }
 }
 
 object RequestMetrics {
   val consumerFetchMetricName = ApiKeys.FETCH.name + "Consumer"
   val followFetchMetricName = ApiKeys.FETCH.name + "Follower"
+
+  val RequestsPerSec = "RequestsPerSec"
+  val RequestQueueTimeMs = "RequestQueueTimeMs"
+  val LocalTimeMs = "LocalTimeMs"
+  val RemoteTimeMs = "RemoteTimeMs"
+  val ThrottleTimeMs = "ThrottleTimeMs"
+  val ResponseQueueTimeMs = "ResponseQueueTimeMs"
+  val ResponseSendTimeMs = "ResponseSendTimeMs"
+  val TotalTimeMs = "TotalTimeMs"
+  val RequestBytes = "RequestBytes"
+  val MessageConversionsTimeMs = "MessageConversionsTimeMs"
+  val TemporaryMemoryBytes = "TemporaryMemoryBytes"
+  val ErrorsPerSec = "ErrorsPerSec"
 }
 
 class RequestMetrics(name: String) extends KafkaMetricsGroup {
   val tags = Map("request" -> name)
-  val requestRate = newMeter("RequestsPerSec", "requests", TimeUnit.SECONDS, tags)
+  val requestRate = newMeter(RequestMetrics.RequestsPerSec, "requests", TimeUnit.SECONDS, tags)
```

Comment:
A common pattern is doing the following at the top of the class:

```scala
import RequestMetrics._
```

Then the class has access to the members of the companion object and we don't need to add `RequestMetrics` everywhere.

Reply:
@ijuma Thank you, updated